agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/resourcemanager/RMAppTracer.java

```diff
@@ -0,0 +1,113 @@
+package com.criteo.hadoop.garmadon.agent.tracers.hadoop.resourcemanager;
+
+import com.criteo.hadoop.garmadon.agent.AsyncEventProcessor;
+import com.criteo.hadoop.garmadon.agent.tracers.ConstructorTracer;
+import com.criteo.hadoop.garmadon.event.proto.ResourceManagerEventProtos;
+import com.criteo.hadoop.garmadon.schema.enums.State;
+import com.criteo.hadoop.garmadon.schema.events.Header;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.Implementation;
+import net.bytebuddy.implementation.SuperMethodCall;
+import net.bytebuddy.implementation.bind.annotation.Argument;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.hadoop.yarn.api.records.ApplicationId;
+import org.apache.hadoop.yarn.server.resourcemanager.RMContextImpl;
+
+import java.lang.instrument.Instrumentation;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import static net.bytebuddy.implementation.MethodDelegation.to;
+import static net.bytebuddy.matcher.ElementMatchers.any;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+
+public class RMAppTracer {
+    public static BiConsumer<Header, Object> eventHandler;
+
+    public static void setup(Header baseHeader, Instrumentation instrumentation, AsyncEventProcessor eventProcessor) {
+        initEventHandler((headerOverride, event) -> {
+            Header header = baseHeader.cloneAndOverride(headerOverride);
+            eventProcessor.offer(header, event);
+        });
+        // Get Events of NEW applications
+        new RMAppImplTracer().installOn(instrumentation);
+        // Get Apps in the RM every 10s
+        new RMContextImplThread().installOn(instrumentation);
+        //new ApplicationMasterServiceTracer().installOn(instrumentation);
+    }
+
+    public static void initEventHandler(BiConsumer<Header, Object> eventHandler) {
+        RMAppTracer.eventHandler = eventHandler;
+    }
+
+    public static class RMContextImplThread extends ConstructorTracer {
+
+        @Override
+        protected ElementMatcher<? super TypeDescription> typeMatcher() {
+            return named("org.apache.hadoop.yarn.server.resourcemanager.RMContextImpl");
+        }
+
+        @Override
+        protected ElementMatcher<? super MethodDescription> constructorMatcher() {
+            return any();
+        }
+
+        @Override
+        protected Implementation newImplementation() {
+            return Advice.to(RMContextImplThread.class);
+        }
+
+        @Advice.OnMethodExit
+        public static void runEventScheduler(@Advice.This Object o) {
+            new ScheduledThreadPoolExecutor(1).scheduleAtFixedRate(
```

Comment:
This thread pool will retain application from stopping. Either add a shutdown hook to shutdown the pool. Consider setting the thread daemon using a threadfactory

ScheduledExecutorService service = new ScheduledThreadPoolExecutor(1, r -> {
            Thread t = new Thread();
            t.setName("GARMADON - App state listing thread");
            t.setDaemon(true);
            return t;
        });

Reply:
Yes I've already pushed an update with appropriate code